app/models/concerns/orchestration/dhcp.rb

```diff
@@ -87,13 +86,21 @@ def boot_server
     bs = tftp.bootServer
     # if that failed, trying to guess out tftp next server based on the smart proxy hostname
     bs ||= URI.parse(subnet.tftp.url).host
-    # now convert it into an ip address (see http://theforeman.org/issues/show/1381)
-    ip = NicIpResolver.new(:nic => self).to_ip_address(bs) if bs.present?
-    return ip unless ip.nil?
 
-    failure _("Unable to determine the host's boot server. The DHCP smart proxy failed to provide this information and this subnet is not provided with TFTP services.")
-  rescue => e
-    failure _("failed to detect boot server: %s") % e, e
+    # only smart proxies with V2 API are supported
+    if subnet.dhcp.capabilities(:DHCP).any? { |c| c == "dhcp_filename_hostname" }
```

Comment:
Isn't this equivalent to:
```suggestion
    if subnet.dhcp.capabilities(:DHCP).include?("dhcp_filename_hostname")
```

Not needed for this PR, but a helper on the smart proxy model would be nice:
```ruby
def has_capability?(feature, capability)
  caps = capabilities(feature)
  caps && caps.include?(capability)
end
```

Reply:
Added the helper.